matrix-utils.el

```diff
@@ -0,0 +1,18 @@
+(defmacro matrix--map-merge (map &rest pairs)
```

Comment:
Yeah, I made this, then I decided to make `matrix--alist` and use that instead.  I left this in in case it proves useful in other places.  I haven't gone through and converted everything to use one of these yet.  I'm not sure if I should even do that, since I think we should probably move toward the API 0.3.0 and client-ng branches, but I think this macro will be useful in them.